presto-main/src/test/java/com/facebook/presto/server/TestQueryStateInfo.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.server;
+
+import com.facebook.presto.execution.QueryInfo;
+import com.facebook.presto.execution.QueryState;
+import com.facebook.presto.execution.QueryStats;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.memory.MemoryPoolId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import org.joda.time.DateTime;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.execution.QueryState.QUEUED;
+import static com.facebook.presto.operator.BlockedReason.WAITING_FOR_MEMORY;
+import static com.facebook.presto.server.QueryStateInfo.createQueryStateInfo;
+import static com.facebook.presto.spi.resourceGroups.ResourceGroupState.CAN_QUEUE;
+import static com.facebook.presto.spi.resourceGroups.ResourceGroupState.CAN_RUN;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static org.testng.Assert.assertEquals;
+
+public class TestQueryStateInfo
+{
+    @Test
+    public void testQueryStateInfo()
+    {
+        ResourceGroupId groupRoot = new ResourceGroupId("root");
+        ResourceGroupId groupRootA = new ResourceGroupId(groupRoot, "a");
+        ResourceGroupId groupRootAX = new ResourceGroupId(groupRootA, "x");
+        ResourceGroupId groupRootAY = new ResourceGroupId(groupRootA, "y");
+        ResourceGroupId groupRootB = new ResourceGroupId(groupRoot, "b");
+
+        ResourceGroupInfo rootAXInfo = new ResourceGroupInfo(
```

Comment:
I thought about that. But there's not really any field that's the same across all groups. I could change the memory limits related fields to be the same but that would only remove 2.